server/src/main/java/org/elasticsearch/repositories/Repository.java

```diff
@@ -152,7 +152,7 @@ void finalizeSnapshot(SnapshotId snapshotId, ShardGenerations shardGenerations,
      * @param listener              completion listener
      */
     void deleteSnapshots(Collection<SnapshotId> snapshotIds, long repositoryStateId, Version repositoryMetaVersion,
-                         ActionListener<Void> listener);
+                         ActionListener<RepositoryData> listener);
```

Comment:
Same as with snapshot finalizations, much easier to keep things consistent (and faster) by passing the repository data through the listeners instead of having to reload it over and over.